src/Microsoft.Azure.WebJobs/ParameterBindingData.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// A reference type for supporting SDK-type bindings in out-of-process Function workers
+    /// </summary>
+    public class ParameterBindingData
+    {
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class with the Content-Type set to application/json</summary>
+        /// <param name="source"> Identifies the extension this event is coming from </param>
+        /// <param name="jsonSerializableData"> An object containing any required information to hydrate an SDK-type object in the out-of-process worker </param>
+        /// <param name="dataSerializationType"> The type to use when serializing the data.
+        /// If not specified, <see cref="object.GetType()"/> will be used on <paramref name="jsonSerializableData"/>.</param>
+        /// <exception cref="ArgumentNullException">Throws when source is null.</exception>
+        /// <exception cref="ArgumentException">Throws when jsonSerializableData is of type BinaryData.</exception>
+        public ParameterBindingData(string source, object jsonSerializableData, Type dataSerializationType = default)
+        {
+            if (jsonSerializableData is BinaryData)
```

Comment:
Wondering if we need this ctor at all, since in the end it's just creating a BinaryData. Why not just have callers create that themselves? Otherwise, we're in the business of doing validation like this, and also responsible for surfacing BinaryData ctor parameters. For example, your ctor doesn't allow them to specify JsonSerializerOptions.

Reply:
Yup, that's a fair point. I guess if they wanted to do it themselves with JsonSerializerOptions they could always use the other ctor. This was just to make it easier but I'm not attached to it so can remove